src/jordan_wigner.rs

```diff
@@ -18,104 +21,163 @@ pub fn jordan_wigner_qiskit(
+                                    multiply_by_zs_and_main(&mut s_y, &z_positions, q, b'Y');
+                                *next.entry(s_y).or_insert(Complex64::new(0.0, 0.0)) +=
+                                    c * coeff_y * phase_y;
+                            }
+                            current = next;
+                        }
+
+                        // Accumulating term contribution into the thread-local accumulator.
+                        for (ps, c) in current.into_iter() {
+                            let w = c * term_coeff;
+                            if w.re.abs() > tol || w.im.abs() > tol {
+                                *acc_local.entry(ps).or_insert(Complex64::new(0.0, 0.0)) += w;
+                            }
+                        }
+                        acc_local
+                    },
+                )
+                .reduce(HashMap::<Vec<u8>, Complex64>::new, |mut a, b| {
+                    for (k, v) in b {
+                        *a.entry(k).or_insert(Complex64::new(0.0, 0.0)) += v;
+                    }
+                    a
+                });
+
+            // Convert dense bytes to compact sparse_list triples
+            let mut sparse_list: SparseList = Vec::with_capacity(acc.len());
+            for (ps_bytes, w) in acc.into_iter() {
+                if w.re.abs() <= tol && w.im.abs() <= tol {
+                    continue;
+                }
+                let mut indices: Vec<usize> = Vec::new();
+                let mut label = String::new();
+                for (q, &ch) in ps_bytes.iter().enumerate() {
+                    if ch != b'I' {
+                        indices.push(q);
+                        label.push(ch as char);
+                    }
+                }
+                // Identity term allowed as ("", [], coeff)
+                sparse_list.push((label, indices, w));
             }
-        }
-        // Identity term allowed as ("", [], coeff)
-        sparse_list.push((label, indices, w));
-    }
+
+            Ok::<_, ()>((sparse_list, n_qubits))
+        })
+    })
+    .map_err(|_| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("JW mapping failed"))?;
 
     Ok((sparse_list, n_qubits))
 }
+
+// Initialize the global Rayon thread-pool once, using env if provided.
+fn init_thread_pool() {
+    static START: Once = Once::new();
+    START.call_once(|| {
+        // 1) RAYON_NUM_THREADS (Rayon's native env var)
+        // 2) OMP_NUM_THREADS (OpenMP-style)
+        // 3) One of the common BLAS-style env vars
+        // 4) Default (Rayon decides)
+        let from_env = |k: &str| env::var(k).ok().and_then(|v| v.parse::<usize>().ok());
+
+        let n_threads = from_env("RAYON_NUM_THREADS")
+            .or_else(|| from_env("OMP_NUM_THREADS"))
+            .or_else(|| from_env("OPENBLAS_NUM_THREADS"))
+            .or_else(|| from_env("MKL_NUM_THREADS"))
+            .or_else(|| from_env("NUMEXPR_NUM_THREADS"));
```

Comment:
I'd rather not handle any environment variables manually. `RAYON_NUM_THREADS` should work regardless, right? If so, that's good enough.